lisa/tools/reboot.py

```diff
@@ -53,11 +52,30 @@ def reboot(self) -> None:
             sleep(wait_seconds)
             current_delta = date.current().replace(tzinfo=None) - current_boot_time
 
+        command_result = self.node.execute(
+            "command -v sudo", shell=True, no_info_log=True
+        )
+        if command_result.exit_code != 0:
+            raise LisaException(
+                "the system doesn't support [command] or [sudo], cannot perform reboot."
+            )
+
+        # Get reboot execution path
+        # Not all distros have the same reboot execution path
+        command_result = self.node.execute(
```

Comment:
try this logic

```python
        command_result = self.node.execute(
            f"sudo -s command -v {self.command}", shell=True, no_info_log=True
        )
        if command_result.exit_code == 0:
            self.command = command_result.stdout
```

Reply:
self.command is read-only, can't be set value.

Reply:
You can create a private member, and set it. Refer to waagent or modinfo.